### ISSUE TITLE
逻辑BUG，将导致分析出错误的main_activity

### ISSUE THREAD

**Issue by BruceFeIix** [MobSF/Mobile-Security-Framework-MobSF#1656]
举例，QQ音乐的APK，静态分析脚本（static_analysis.py）执行到下面逻辑代码时，将取出错误的mainact:
'''
            if len(mainact) < 1:
                for sitem in activity.getElementsByTagName('action'):
                    val = sitem.getAttribute('android:name')
                    if val == 'android.intent.action.MAIN':
                        mainact = activity.getAttribute('android:name')
                if mainact == '':
                    for sitem in activity.getElementsByTagName('category'):
                        val = sitem.getAttribute('android:name')
                        if val == 'android.intent.category.LAUNCHER':
                            mainact = activity.getAttribute('android:name')
'''

触发BUG的原始输入数据如下：
`
<activity android:theme="@style/ih" android:name="com.tencent.qqmusic.activity.ShareAccountActivity" android:exported="false" android:launchMode="singleTop" android:screenOrientation="portrait" android:configChanges="keyboardHidden|orientation|screenLayout|screenSize|smallestScreenSize" android:windowSoftInputMode="adjustPan">
            <intent-filter>
                <action android:name="com.tencent.qqmusiccommon.PLAYLIST_VIEWER.PHONE" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        <activity android:theme="@style/i4" android:name="com.tencent.qqmusic.activity.AppStarterActivity" android:launchMode="singleTop" android:screenOrientation="portrait" android:configChanges="keyboardHidden|mcc|mnc|orientation|screenLayout|screenSize|smallestScreenSize|uiMode" android:windowSoftInputMode="adjustPan">
            <intent-filter>
                <action android:name="com.tencent.qqmusiccommon.PLAYLIST_VIEWER.PHONE" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
            <meta-data android:name="android.app.shortcuts" android:resource="@xml/h" />
        </activity>
`

如上所述，该代码执行至第一个ShareAccountActivity后，由于未检测到val == 'android.intent.action.MAIN'，紧接着会执行后续的检测逻辑if val == 'android.intent.category.LAUNCHER':，将会为mainact 赋值为当前ShareAccountActivity。

建议：
移除后续的冗余代码，单纯通过'android.intent.action.MAIN'来检测main_activity，或者采取更完善的检测逻辑。

'''
            if len(mainact) < 1:
                # ^ Fix for Shitty Manifest with more than one MAIN
                for sitem in activity.getElementsByTagName('action'):
                    val = sitem.getAttribute('android:name')
                    if val == 'android.intent.action.MAIN':
                        mainact = activity.getAttribute('android:name')
                # remove this code
                # if mainact == '':
                #     for sitem in activity.getElementsByTagName('category'):
                #         val = sitem.getAttribute('android:name')
                #         if val == 'android.intent.category.LAUNCHER':
                #             mainact = activity.getAttribute('android:name')
'''

**Comment by ajinabraham** [comment]
Thanks for reporting.

Translating 

For example, when the QQ Music APK and static analysis script (static_analysis.py) are executed to the following logic code, the wrong mainact will be retrieved:
```
if len(mainact) <1:
for sitem in activity.getElementsByTagName('action'):
val = sitem.getAttribute('android:name')
if val =='android.intent.action.MAIN':
mainact = activity.getAttribute('android:name')
if mainact =='':
for sitem in activity.getElementsByTagName('category'):
val = sitem.getAttribute('android:name')
if val =='android.intent.category.LAUNCHER':
mainact = activity.getAttribute('android:name')
```

The original input data that triggers the BUG are as follows:

```
<activity android:theme="@style/ih" android:name="com.tencent.qqmusic.activity.ShareAccountActivity" android:exported="false" android:launchMode="singleTop" android:screenOrientation="portrait" android: configChanges="keyboardHidden|orientation|screenLayout|screenSize|smallestScreenSize" android:windowSoftInputMode="adjustPan"> <intent-filter> <action android:name="com.tencent.qqmusiccommon.PLAYLIST_VIEWER.PHONE" /> <category android:name ="android.intent.category.DEFAULT" /> <category android:name="android.intent.category.LAUNCHER" /> </intent-filter> </activity> <activity android:theme="@style/i4 "android:name="com.tencent.qqmusic.activity.AppStarterActivity" android:launchMode="singleTop" android:screenOrientation="portrait" android:configChanges="keyboardHidden|mcc|mnc|orientation|screenLayout|screenSize|smallestScreenSize|uiMode "android:windowSoftInputMode="adjustPan"> <intent-filter> <action android:name="com.tencent.qqmusiccommon.PLAYLIST_VIEWER.PHONE" /> <category android:name=" android.intent.category.DEFAULT" /> </intent-filter> <intent-filter> <action android:name="android.intent.action.MAIN" /> <category android:name="android.intent.category .LAUNCHER" /> </intent-filter> <intent-filter> <action android:name="android.intent.action.VIEW" /> <category android:name="android.intent.category.DEFAULT" /> </intent-filter> <meta-data android:name="android.app.shortcuts" android:resource="@xml/h" /> </activity>
```

As mentioned above, after the code is executed to the first ShareAccountActivity, since val =='android.intent.action.MAIN' is not detected, the subsequent detection logic if val =='android.intent.category will be executed. LAUNCHER': will assign the current ShareAccountActivity to mainact.

Suggest:
Remove the follow-up redundant code, and simply use'android.intent.action.MAIN' to detect main_activity, or adopt a more complete detection logic.

```
if len(mainact) <1:
# ^ Fix for Shitty Manifest with more than one MAIN
for sitem in activity.getElementsByTagName('action'):
val = sitem.getAttribute('android:name')
if val =='android.intent.action.MAIN':
mainact = activity.getAttribute('android:name')
# remove this code
# if mainact =='':
# for sitem in activity.getElementsByTagName('category'):
# val = sitem.getAttribute('android:name')
# if val =='android.intent.category.LAUNCHER':
# mainact = activity.getAttribute('android:name')
```

**Comment by ajinabraham** [comment]
This is fixed in latest master (3.3.0)
